paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/action/SortCompactAction.java

```diff
@@ -116,16 +116,11 @@ public void build(StreamExecutionEnvironment env) {
         DataStream<RowData> source = sourceBuilder.withEnv(env).withContinuousMode(false).build();
         TableSorter sorter =
                 TableSorter.getSorter(env, source, fileStoreTable, sortStrategy, orderColumns);
-        DataStream<RowData> sorted = sorter.sort();
 
-        FlinkSinkBuilder flinkSinkBuilder = new FlinkSinkBuilder(fileStoreTable);
-        flinkSinkBuilder.withInput(sorted).withOverwritePartition(new HashMap<>());
-        String sinkParallelism = tableConfig.get(FlinkConnectorOptions.SINK_PARALLELISM.key());
-        if (sinkParallelism != null) {
-            flinkSinkBuilder.withParallelism(Integer.parseInt(sinkParallelism));
-        }
-
-        flinkSinkBuilder.build();
+        new FlinkSinkBuilder(fileStoreTable)
+                .withInput(sorter.sort())
+                .withOverwritePartition(new HashMap<>())
```

Comment:
Could the `overwritePartition` of `withOverwritePartition` be nullable?

Reply:
sure

Reply:
When `overwritePartition` is nullable, does this need to invoke `withOverwritePartition`?

Reply:
Yes, it need to tag it as overwrite. If we don't call withOverwritePartition(new HastMap()), it will not be a compact and overwrite action, the data will be doubled

Reply:
See `FlinkWriteSink` constructor.